android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +779,98 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
         return motionEventList;
     }
 
+    public VideoStreamingParameters getLastCachedStreamingParameters() {
+        return parameters;
+    }
+
+    private List<VideoStreamingCapability> getSupportedCapabilities(
+            List<VideoStreamingRange> ranges,
+            VideoStreamingCapability rootCapability
+    ){
+
+        List<VideoStreamingCapability> validCapabilities = new ArrayList<>();
+        List<VideoStreamingCapability> allCapabilities = rootCapability.getAdditionalVideoStreamingCapabilities();
+        allCapabilities.add(rootCapability);
+
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+            if (allCapabilities != null && !allCapabilities.isEmpty()) {
+
+                for (VideoStreamingCapability capability : allCapabilities) {
+                    double diagonal;
+                    if (capability.getPreferredResolution() == null
+                            || capability.getPreferredResolution().getResolutionHeight() == null
+                            || capability.getPreferredResolution().getResolutionWidth() == null) {
+                        continue;
+                    }
+                    if (capability.getDiagonalScreenSize() == null) {
+                        diagonal = parameters.getPreferredDiagonal();
+                    } else {
+                        diagonal = capability.getDiagonalScreenSize();
+                    }
+
+                    if (range.getMinScreenDiagonal() > diagonal) {
+                        continue;
+                    }
+
+                    if (!isAspectRatioInRange(range.getMinAspectRatio(), range.getMaxAspectRatio(), capability.getPreferredResolution())) {
+                        if (constraintHeightMax == null && constraintHeightMin == null) {
+                            continue;
+                        }
+                    }
+
+                    if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                        continue;
+                    }
+
+                    if (!validCapabilities.contains(capability)) {
```

Comment:
```suggestion
                    capability.setAdditionalVideoStreamingCapabilities(null);
                    if (!validCapabilities.contains(capability)) {
```
Hello @kostyaBoss to avoid the edge case I discussed earlier from happening within the manager and to align to the iOS implementation we should set the additionalCapability of any capability to be added to null before it is added.